### DISCUSSION TITLE
CORS error while using Laravel Cashier

### DISCUSSION THREAD

**Discussion by franklin-tina** [inertiajs/inertia#1150] [Help]
Hi, I know this is an old issue, and I have seen **use `Inertia::location`** on so many proposed solutions.
For example, [this](https://github.com/inertiajs/inertia/discussions/871) is a similar issue.

But In my case, I'm using the `checkoutCharge` method provided by laravel cashier for Stripe one time payments like so:
```php
 return $user->checkoutCharge($totalAmount, "SomeProductName", 1, [
    'success_url' => route('products.checkout-success') . '?session_id={CHECKOUT_SESSION_ID}',
    'cancel_url' => route('products.checkout-cancel'),
]);
```
How will I overcome the **cors** error?

### My environment and packages used:
- Laravel 9.2
- Vue 3
- Laravel cashier 13.8
- MacOS 12

**Comment by mbrzuzy** [comment]
Hey,

A bit late, but this might help someone in the future. Try the following. `checkoutCharge` is calling `checkout` which should return an object with a url that you can use to redirect and overcome the CORs issue.

```
return Inertia::location($user->checkoutCharge($totalAmount, "SomeProductName", 1, [
    'success_url' => route('products.checkout-success') . '?session_id={CHECKOUT_SESSION_ID}',
    'cancel_url' => route('products.checkout-cancel'),
])->url);
```
